stactools_sentinel1/stactools/sentinel1/rtc_metadata.py

```diff
@@ -0,0 +1,166 @@
+from typing import List, Optional
+
+import pystac
+from pystac.utils import str_to_datetime
+
+import rasterio
+import geopandas as gpd
+import os
+import io
+import json
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class RTCMetadata:
+    def __init__(self, href):
+        self.href = href
+
+        def _load_metadata(asset='local_incident_angle.tif'):
+            ''' key metadata stored in Geotiff tags '''
+            with rasterio.Env(AWS_NO_SIGN_REQUEST='YES',
+                              GDAL_DISABLE_READDIR_ON_OPEN='EMPTY_DIR'):
+                with rasterio.open(os.path.join(href, asset)) as src:
+                    metadata = src.profile
+                    metadata.update(src.tags())
+                    # other things that aren't keys in src.profile
+                    metadata['PROJ_BBOX'] = list(src.bounds)
+                    metadata['SHAPE'] = src.shape
+
+            return metadata
+
+        def _get_geometries():
+            ''' bbox is MGRS grid square, but geometry is intersection of bbox
+            and S1 frames use to create RTC product. EPSG:4326'''
+            # NOTE: could also save proj:geometry in UTM CRS...
+            gridfile = os.path.join(os.path.dirname(__file__),
+                                    'sentinel1-rtc-conus-grid.geojson')
```

Comment:
I think this file contains the local UTM geometry too, so you wouldn't have to compute it.

Reply:
no longer using this auxiliary file since it is much more accurate to extract footprints with rasterio+shapely